### PR TITLE
fix: reset to default settings if healthCheck is removed (#369)

### PR DESCRIPTION
**Pull request by scottlaiaws**
**What type of PR is this?**
bug

**Which issue does this PR fix**:
#369 

**What does this PR do / Why do we need it**:
Fix bug #369 by resetting the health check configuration to default setting if `healthCheck` section is removed

**If an issue # is not available please add repro steps and logs from aws-gateway-controller showing the issue**:

**Testing done on this change**:
Manual testing, confirmed health check configuration is reset if removed. Also ran e2e tests

```
Ran 31 of 31 Specs in 2030.679 seconds
SUCCESS! -- 31 Passed | 0 Failed | 0 Pending | 0 Skipped
--- PASS: TestIntegration (2031.56s)
PASS
ok      github.com/aws/aws-application-networking-k8s/test/suites/integration   2031.595s
```

**Automation added to e2e**:
N/A

**Will this PR introduce any new dependencies?**: No

**Will this break upgrades or downgrades. Has updating a running cluster been tested?**:

**Does this PR introduce any user-facing change?**: Yes

```release-note
fix bug #369 
```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by zijun726911** [comment]
QQ: if we delete the whole TargetGroupPolicy k8s resource, can it also reset to the default healthCheck?

File: pkg/deploy/lattice/target_group_manager.go [hunk 1]
```diff
@@ -382,3 +411,44 @@ func (s *defaultTargetGroupManager) findTargetGroup(ctx context.Context, targetG
...
+	defaultProtocol := vpclattice.TargetGroupProtocolHttp
+
+	isTgProtocolVersionEmptyOrGrpc := targetGroupProtocolVersion == "" || targetGroupProtocolVersion == vpclattice.TargetGroupProtocolVersionGrpc
+
+	if isTgProtocolVersionEmptyOrGrpc {
+		targetGroupProtocolVersion = vpclattice.HealthCheckProtocolVersionHttp1
+	}
+
+	enabled := targetGroupProtocolVersion == vpclattice.TargetGroupProtocolVersionHttp1
+
```
Comment (zijun726911) [inline pkg/deploy/lattice/target_group_manager.go:435]: Seems from current code logic, the Grpc defauit HealthCheckConfig Enabled==true? I think it should be false? 



Would be good if you can also a unit test for getDefaultHealthCheckConfig

**Comment by scottlaiaws** [comment]
> QQ: if we delete the whole TargetGroupPolicy k8s resource, can it also reset to the default healthCheck?

I just tested locally. Yes, it would reset to default health check configuration

File: pkg/deploy/lattice/target_group_manager.go (hunk 1 above)
Comment (scottlaiaws) [reply pkg/deploy/lattice/target_group_manager.go:435]: thanks for the catch, I'll fix this, and add unit tests

**Review by zijun726911 — APPROVED** [review]
LGTM, Do you have a time to take a look? @solmonk

File: pkg/deploy/lattice/target_group_manager.go [hunk 2]
```diff
@@ -382,3 +411,48 @@ func (s *defaultTargetGroupManager) findTargetGroup(ctx context.Context, targetG
...
+	if targetGroupProtocolVersion == vpclattice.TargetGroupProtocolVersionGrpc {
+		healthCheckProtocolVersion = vpclattice.HealthCheckProtocolVersionHttp1
+	}
+
+	return &vpclattice.HealthCheckConfig{
+		Enabled: &enabled,
+
+		// By default use port of the target group
+		Port: &targetGroupPort,
```
[the remark this answers is not in the document]
Comment (solmonk) [reply pkg/deploy/lattice/target_group_manager.go:444]: also curious on why we are "resetting" some fields but putting default values on some other fields.

[the remark this answers is not in the document]
Comment (scottlaiaws) [reply pkg/deploy/lattice/target_group_manager.go:444]: oh you're right, I need to confirm if setting to `resetValue` (i.e. `0`) would do that

**Pull request closed without merging**